core/src/test/java/org/elasticsearch/indices/cluster/IndicesClusterStateServiceRandomUpdatesTests.java

```diff
@@ -97,6 +100,49 @@ public void testRandomClusterStateUpdates() {
         logger.info("Final cluster state: {}", state.prettyPrint());
     }
 
+    public void testSimulateMasterDataDirectoryWiped() {
+        // we have an IndicesClusterStateService per node in the cluster
+        final Map<DiscoveryNode, IndicesClusterStateService> clusterStateServiceMap = new HashMap<>();
+        ClusterState previousState = randomInitialClusterState(clusterStateServiceMap);
+        ClusterState newClusterState = ClusterState.builder(previousState)
+                                           .metaData(MetaData.builder(previousState.metaData()).clusterUUID(UUIDs.randomBase64UUID()))
+                                           .build();
+
+        // add index to the cluster
+        String name = "index_" + randomAsciiOfLength(8).toLowerCase(Locale.ROOT);
+        CreateIndexRequest request = new CreateIndexRequest(name, Settings.builder()
+                                                                      .put(SETTING_NUMBER_OF_SHARDS, randomIntBetween(1, 3))
+                                                                      .put(SETTING_NUMBER_OF_REPLICAS, 1)
+                                                                      .build()).waitForActiveShards(ActiveShardCount.ALL);
```

Comment:
does this work? I thought that waitForActiveShards would not work with the mock ClusterChanges class?